test/cases/cross-distro.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/bash
+
+#
+# Test the available distributions. Only allow releases for the current distro.
+#
+
+APISOCKET=/run/weldr/api.socket
+
+source /etc/os-release
+
+# Build a grep pattern that results in an empty string when the expected distros are installed
+case $ID in
+    fedora)
+        PATTERN="\[|\]|fedora-"
+        ;;
+    rhel)
+        MAJOR=$(echo "$VERSION_ID" | sed -E 's/\..*//')
+        case $MAJOR in
+            8)
+                # RHEL 8 only supports building RHEL 8
+                PATTERN="\[|\]|rhel-$MAJOR"
+                ;;
+            *)
+                # RHEL 9 and later support building all releases
+                PATTERN="\[|\]|rhel-*"
+                ;;
+        esac
+        ;;
+    centos)
+        MAJOR=$(echo "$VERSION_ID" | sed -E 's/\..*//')
+        case $MAJOR in
+            8)
+                # CentOS 8 only supports building CentosOS 8
+                PATTERN="\[|\]|centos-$MAJOR"
+                ;;
+            *)
+                # CentOS 9 and later support building all releases
+                PATTERN="\[|\]|centos-*"
+                ;;
+        esac
+        ;;
+    *)
+        echo "Unknown distribution id: $ID 😢"
+        exit 1
+    ;;
+esac
+
+
+# Provision the software under test.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+# composer-cli in RHEL 8 doesn't support distro command, so use curl for this test
+if [ ! -e $APISOCKET ]; then
+    echo "osbuild-composer.socket has not been started. 😢"
+    exit 1
+fi
+
+if ! curl -s --unix-socket $APISOCKET http:///localhost/api/status > /dev/null; then
+    echo "osbuild-composer server not available. 😢"
+    exit 1
+fi
+
+if ! DISTROS=$(curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
+    echo "osbuild-composer server error getting distros list. 😢"
+    exit 1
+fi
+
+REMAINDER=$(echo "$DISTROS" | jq -r '.distros' | grep -v -E "$PATTERN")
+if [ -n "$REMAINDER" ]; then
+	echo "🔥 Unexpected distros installed:"
```

Comment:
Indentation error?

Reply:
yeah, some tabs snuck back in. should be fixed now.